### ISSUE TITLE
Testcontainers Golang module for WireMock

### ISSUE THREAD

**Issue by oleg-nenashev** [wiremock/ecosystem#9]
### Proposal

Similar to Java, it would be nice to add support for starting the WireMock server using Testcontainers when Docker is present and no other WireMock server has been provided

### If there are any existing implementations, prototypes, blogs or etc, let us know!

There is WireMock Go implementation based on WireMock Docker, but not a Testcontainers one. See the [WireMock Ecosystem listing](https://github.com/wiremock/ecosystem) for the references

### Reference any relevant documentation, other materials or issues/pull requests that can be used for inspiration.

- Java module: https://github.com/wiremock/wiremock-testcontainers-java
- https://golang.testcontainers.org/

**Comment by kapishmalik** [comment]
@oleg-nenashev can I work on this?

**Comment by oleg-nenashev** [comment]
@kapishmalik If you are interested, sure, go for it. Note that there might be some functionality overlap with https://github.com/walkerus/go-wiremock , and it would be great to collaborate with @walkerus on unifying/sharing APIs and maybe even co-hosting components. I reached out to him on LinkedIn

**Comment by kapishmalik** [comment]
@oleg-nenashev yup, true. I was also thinking the same. He has already written wrapper. We just need to add functionality to invoke testcontainer in that wrapper.

**Comment by oleg-nenashev** [comment]
PoC is here: https://github.com/wiremock/wiremock-testcontainers-go
